lib/app/features/home/home_widgets/home_options_tap_bar/home_options_tap_bar.dart

```diff
@@ -12,67 +12,44 @@ class HomeOptionsTapBar extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Padding(
-      padding: const EdgeInsets.symmetric(vertical: 8.0),
-      child: Row(
-        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-        children: [
-          GestureDetector(
-            onTap: () {
-              _homeCubit(context).switchHomeOptions(HomeOptions.upcoming);
-            },
-            child: Container(
-              padding: const EdgeInsets.only(bottom: 4.0),
-              decoration: const BoxDecoration(
-                  border: Border(
-                      bottom: BorderSide(
-                width: 3.0,
-                color: AppColors.mainThemePink,
-              ))),
-              child: Text(
-                context.localizations.homeUpcoming,
-                style: const TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
+    return BlocBuilder<HomeCubit, HomeState>(
+      builder: (context, state) {
```

Comment:
Consider adding buildWhen to rebuild only when necessary

Reply:
done